cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -253,11 +253,11 @@ TEST_F(TestIsInKernel, TimeDuration) {
               "[true, false, false, true, true]", /*skip_nulls=*/true);
   }
 
-  // Different units, invalid cast
-  ASSERT_RAISES(Invalid, IsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 1, 2]"),
-                              ArrayFromJSON(duration(TimeUnit::MILLI), "[0, 2]")));
+  // Different units, cast value_set to values
+  CheckIsIn(ArrayFromJSON(duration(TimeUnit::SECOND), "[0, 2]"),
+            ArrayFromJSON(duration(TimeUnit::MILLI), "[0, 1, 2000]"), "[true, true]");
```

Comment:
Ok, but you're choosing an easy case here.
What if values = `[0s, 2s]` and value_set = `[1ms, 2ms, 2000ms]`? Will it fail because of truncation during casting? Will it return an incorrect value? Will it try casting the other way round?

Reply:
Updated the test case. Since only safe casts are allowed, it will first try casting [1ms, 2ms, 2000ms] to [0s, 2s] and fail, then cast the other way around and succeed. The result is [null, true] as expected.